packages/ifpack2/src/Ifpack2_RILUK_decl.hpp

```diff
@@ -640,10 +640,9 @@ clone (const Teuchos::RCP<const NewMatrixType>& A_newnode) const
   RCP<ParameterList> plClone = Teuchos::parameterList ();
   plClone = detail::setLocalSolveParams<NewMatrixType, new_node_type>::setParams (plClone);
 
-  RCP<new_node_type> new_node = A_newnode->getNode ();
-  new_riluk->L_ = L_->clone (new_node, plClone);
-  new_riluk->U_ = U_->clone (new_node, plClone);
-  new_riluk->D_ = D_->clone (new_node);
+  new_riluk->L_ = rcp(new crs_matrix_type(L_, Teuchos::Copy));
```

Comment:
The deep copy constructor `CrsMatrix(const CrsMatrix&, Teuchos::Copy)` currently only works if the input CrsMatrix is fill complete.  It would be easy to fix that, but please do be aware of that for now.  Thanks!

Reply:
Noted; I think it is fine here.